scipy/signal/_signaltools.py

```diff
@@ -243,13 +247,24 @@ def correlate(in1, in2, mode='full', method='auto'):
     >>> plt.show()
 
     """
-    in1 = np.asarray(in1)
-    in2 = np.asarray(in2)
-    _reject_objects(in1, 'correlate')
-    _reject_objects(in1, 'correlate')
+    try:
+        xp = array_namespace(in1, in2)
+    except TypeError:
+        # either in1 or in2 are object arrays
+        xp = np_compat
```

Comment:
what about other disallowed array types like `numpy.matrix` and masked arrays - do we end up with more cryptic error messages here?

Reply:
Let's start worrying about this when it actually happens.